### ISSUE TITLE
Should deserialization silently pass on mismatching arguments?

### ISSUE THREAD

**Issue by cypreess** [ramonhagenaars/jsons#12]
Let's consider the following example:
```
In [30]: @dataclass
    ...: class Test:
    ...:     a: int
    ...:     b: int
    ...:

In [31]: Test(1,2)
Out[31]: Test(a=1, b=2)

In [32]: jsons.dumps(Test(1,2))
Out[32]: '{"a": 1, "b": 2}'

In [33]: jsons.loads('{"a": 1, "b": 2, "c": 3}', Test)
Out[33]: Test(a=1, b=2)
```

I would expect jsons to fail in the last statement, as the protocol is not strictly the same.

Is there any rationale for such behavior? If yes, that would be great to have a `strict=True` argument for jsons.

**Comment by ramonhagenaars** [comment]
Thank you cypreess, for your issue.

The thoughts on this are the following:
* `jsons` was designed to be very tolerant by default.
* `jsons` was in fact capable of deserializing that json data into `Test`; all required fields were provided. So it is reasonable that no error should occur.
* `jsons` should be compatible with json schemas and they allow extra attributes by default as well.

Having said that, I can see how your addition could be useful. `jsons` does actually already have a `strict: bool` parameter. Your suggestion will be added to that "strict-mode" behavior.

This feature will be included in the `0.7` release.

**Comment by ramonhagenaars** [comment]
Fixed in [0.7.0](https://pypi.org/project/jsons/0.7.0/).
